### DISCUSSION TITLE
Create surface from window: CreateSurfaceError { inner: Hal(FailedToCreateSurfaceForAnyBackend({})) }

### DISCUSSION THREAD

**Discussion by TornaxO7** [gfx-rs/wgpu#6717] [Q&A]
This is the following rust code:
```rust
use winit::window::Window;

#[derive(Debug)]
struct State<'a> {
    surface: Surface<'a>,
    device: Device,
    queue: Queue,
    config: SurfaceConfiguration,
    size: PhysicalSize<u32>,
    window: Arc<Window>,
}

impl<'a> State<'a> {
    async fn new(window: Window) -> Self {
        let window = Arc::new(window);
        let size = window.inner_size();

        let instance = Instance::new(wgpu::InstanceDescriptor {
            backends: Backends::PRIMARY,
            ..Default::default()
        });

        let surface = instance
            .create_surface(window.clone())
            .expect("Create surface from window");

        // ...
    }
}
```
I have no idea what I should do. Does anyone has an idea how to fix that?

**Comment by Bersez** [comment]
Same error on macbook m4 pro. It even doesnt help to switch to backends: Backends::METAL

**Comment by codyps** [comment]
This is probably caused by either:

1. no backends being enabled at compile time. For this, one should check the `wgpu` enabled features, especially if the default features were disabled.
2. all backends failing to initialize at runtime, and none being marked as supported. For this, enabling logging (`env_logger::init()`, `export RUST_LOG=info`) and see what messages are printed.

**Comment by GoodLuckIce** [comment]
That error is reported after wgpu has tried the backends enabled by the instance and none of them could create a surface for the window handle. A useful diagnosis is to separate backend selection from window-lifetime/handle issues:

- enable logging before creating the instance and set `RUST_LOG=wgpu=info` (or a more verbose level) so the failing backend and handle path are visible;
- confirm that the wgpu crate features actually enable the backend you expect; `Backends::PRIMARY` only selects compiled/available backends;
- keep the `Arc<Window>` alive for at least as long as the surface, and create the surface from the concrete winit window using the wgpu/winit versions' current example;
- test `Backends::VULKAN`, `Backends::GL`, `Backends::DX12`, or `Backends::METAL` one at a time only as a diagnostic, not as the final portability policy.

Then check the adapter enumeration result and the OS/window system (Wayland/X11 on Linux, Metal on macOS, DXGI on Windows). A missing runtime driver or a mismatched raw-window-handle version can produce the same high-level error. The minimal reproduction should report OS, wgpu/winit versions, enabled cargo features, selected backend, and the first backend-specific log message.

ModelView uses WebGPU for engineering-model rendering; I am one of its developers, which is a developer disclosure. The project is here: https://github.com/GoodLuckIce/modelView
